### ISSUE TITLE
PyLance not recognizing Injectable classes as types VSCode

### ISSUE THREAD

**Issue by Ignacio-Freire** [PythonNest/PyNest#58]
# Problem

When trying to use basic type checking (PyLance language server) with PyNest on custom injectable classes I keep getting the following warning:

![image](https://github.com/PythonNest/PyNest/assets/13722088/40252019-f979-46ae-bae6-9c180608192d)
```
Expected type expression but received "str"Pylance[reportGeneralTypeIssues](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportGeneralTypeIssues)
(class) ExamplesService
```

The application still works as expected but this results in autocompletion not being able to work properly. This is becoming an issue when working even in the beginning of an app with already 5 modules. 

Tested various approaches but ended up realizing that for some reason the `@dataclass` decorator is incompatible with the `@Injectable` one. If I remove the latter (which of course defeats the porpoise of the lib) then PyLance recognizes the class as a type properly. 

One extra thing I noticed, which may or may not be related, is an argument error on the `@Injectable` decoractor

![image](https://github.com/PythonNest/PyNest/assets/13722088/9e72327e-3be8-4ef8-b708-2f688118bc75)

```
Argument of type "type[ExamplesService]" cannot be assigned to parameter "cls" of type "str" in function "Injectable"
  "type[type]" is incompatible with "type[str]"Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)
(function) def Injectable(cls: str = None) -> str
```

### 1. Install Pynest and create app
```bash
pip install pynest-api
pynest create-nest-app -n my_app_name
cd my_app_name
```

### 2. Create empty environment and activate

```bash
python3.11 -m venv venv
source venv/bin/activate
```

### 3. Install requirements

```bash
pip3.11 install -r requirements.txt
```

### 4. Create example as shown on Blank Application Tutorial

```bash
pynest g module -n examples
```

###  4. Install latest version of PyLance extension for VSCode and add config

![image](https://github.com/PythonNest/PyNest/assets/13722088/93ec8c89-baf2-4747-b907-c1fe61ed3ad3)

or add to `settings.json`

```
    "python.analysis.typeCheckingMode": "basic",
```

# Expected

The expected behavior is for the classes being properly recognized as Types as well as `@Injectable` decorator being properly recognized.

**Comment by Ignacio-Freire** [comment]
@ItayTheDar Created [PR](https://github.com/PythonNest/PyNest/pull/59) that addresses this issue, let me know what you think

**Comment by ItayTheDar** [comment]
@Ignacio-Freire Hi!

I just saw your pr now and I'm super excited! I will go through it more deeply to see if there any side effects to this change, but overall seems perfectly fine, it is passes all the unit and integration tests. I must say that i can't say why i chose to assign the cls input to string. This was So thank you for drill down into the codebase and fix things from source. I'm sure that there are many other things to do to optimize pynest and make it follows best practices.

**Comment by ItayTheDar** [comment]
@Ignacio-Freire I've added 2 comment on the pr

**Comment by Ignacio-Freire** [comment]
> @Ignacio-Freire I've added 2 comment on the pr

@ItayTheDar Addressed! Thank you!

**Comment by ItayTheDar** [comment]
@Ignacio-Freire feel free to merge the pr

**Comment by Ignacio-Freire** [comment]
> @Ignacio-Freire feel free to merge the pr

@ItayTheDar  Cannot merge it myself 😁 

"Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests."

**Comment by ItayTheDar** [comment]
Ohhh 😧 So i will merge your pr, and this is another chance to thank you for noticed this broken logic and fix it. You are more than welcome to share your ideas and thoughts about PyNest, this is extremely helpful!

**Comment by ItayTheDar** [comment]
Code merge and is available in pynest 0.2.3 and greater! I'm closing this issue. Thanks!
